packages/react-aria-components/src/Table.tsx

```diff
@@ -102,7 +102,8 @@ class TableCollection<T> extends BaseCollection<T> implements ITableCollection<T
 
   getLastKey() {
     let rows = [...this.getChildren(this.body.key)];
-    return rows[rows.length - 1]?.key;
+    let lastRow = rows[rows.length - 1];
+    return (lastRow?.type === 'loader' ? lastRow.prevKey : lastRow?.key) as Key;
```

Comment:
not sure why typescript insists that this return is different from BaseCollection's (somehow BaseCollection's return type is Key and not Key | null)